relations/jf/src/note.rs

```diff
@@ -0,0 +1,167 @@
+use ark_ff::Zero;
+use jf_primitives::circuit::rescue::RescueNativeGadget;
+use jf_relation::{Circuit, PlonkCircuit, Variable};
+
+use crate::{
+    shielder_types::{convert_hash, Note, Nullifier, TokenAmount, TokenId, Trapdoor},
+    CircuitField, PlonkResult, PublicInput,
+};
+
+pub enum NoteType {
+    Deposit,
+    Spend,
+}
+
+pub struct SourcedNote {
```

Comment:
What does _sourced_ mean?

Reply:
my naming, sorry - "note" is just a hash; "sourced note" is a hash with its source, i.e. preimage